index.js

```diff
@@ -1,17 +1,34 @@
+
 import isObservable from 'is-observable';
 import symbolObservable from 'symbol-observable';
 
-export default async function observableToPromise(value) {
+/**
+ * @template T
+ * @param {T extends unknown<T>} value
+ * @param {{maximumValues?: number}} options
+ * @returns {Promise<T[]>}
+ */
+export default async function observableToPromise(value, options) {
+	const {maximumValues = 0} = options || {};
+
 	if (!isObservable(value)) {
 		throw new TypeError(`Expected an \`Observable\`, got \`${typeof value}\``);
 	}
 
 	const values = [];
+	let count = 0;
 
 	return new Promise((resolve, reject) => {
 		value[symbolObservable.default]().subscribe({
 			next(value) {
-				values.push(value);
+				if (maximumValues > 0 && count < maximumValues) {
+					values.push(value);
+					count += 1;
+				}
+
+				if (maximumValues === 0) {
+					values.push(value);
+				}
```

Comment:
When you reach max count, shouldn't you somehow end the observable stream? Otherwise, it will wastefully continue emitting events even if we don't use them.

Reply:
Thats a big challenge since the libraries don't have a common standard to end the stream, although the logic is very similar. I believe that leaves us with the option to check which library is being used before ending the stream in a different way for each library or slice the observable before subscribing.

To this moment I have used pipes for rxjs but I'm still thinking in a common way for all the libraries.